zlib-rs/src/crc32.rs

```diff
@@ -0,0 +1,133 @@
+const CRC32_LSB_POLY: usize = 0xedb8_8320usize;
+const START_CHECKSUM: u32 = 0xffff_ffffu32;
+const FINAL_XOR: u32 = 0xffff_ffffu32;
+
+const W: usize = std::mem::size_of::<usize>();
+const N: usize = 5;
+
+static CRC32_BYTE_TABLE: [[u32; 256]; 1] = build_crc32_table(1);
+static CRC32_WORD_TABLE: [[u32; 256]; W] = build_crc32_table(1);
+static CRC32_STRIDE_TABLE: [[u32; 256]; W] = build_crc32_table(N);
```

Comment:
these could use some comments 

- you mentioned crc32 using different polynomials, maybe say something about the choice here (though I guess the png just says: we're using this one).
- what is `N`?
- how are these tables used